File: 804. Unique Morse Code Words.py
class Solution:
    def uniqueMorseRepresentations(self, words):
        morseForAtoZ = [
            ".-",
            "-...",
            "-.-.",
            "-..",
            ".",
            "..-.",
            "--.",
            "....",
            "..",
            ".---",
            "-.-",
            ".-..",
            "--",
            "-.",
            "---",
            ".--.",
            "--.-",
            ".-.",
            "...",
            "-",
            "..-",
            "...-",
            ".--",
            "-..-",
            "-.--",
            "--..",
        ]
        aToZ = [
            "a",
            "b",
            "c",
            "d",
            "e",
            "f",
            "g",
            "h",
            "i",
            "j",
            "k",
            "l",
            "m",
            "n",
            "o",
            "p",
            "q",
            "r",
            "s",
            "t",
            "u",
            "v",
            "w",
            "x",
            "y",
            "z",
        ]
        newDict = {}
        for i, j in zip(morseForAtoZ, aToZ):
            newDict[i] = j
        print(newDict)


newUniqueMorse = Solution()
newUniqueMorse.uniqueMorseRepresentations(["gin", "zen", "gig", "msg"])

File: 804_unique-morse-code-words.c
#include <stdio.h>

int uniqueMorseRepresentations(char **words, int wordsSize)
{
    char engAlphabet[26][5] = {".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."};

    for (int i = 0; i < wordsSize; i++)
    {
        // printf("%s ", words[i]);
        printf("%s ", engAlphabet[*words[i] - 97]);
    }

    // printf("%d", wordsSize);
    // printf("%s", words[0]);
}

int main()

{
    char words1[] = "gin";
    char words2[] = "zen";
    char words3[] = "gig";
    char words4[] = "msg";

    char *words[] = {words1, words2, words3, words4};
    int wordsSize = sizeof(words) / sizeof(words[0]);

    uniqueMorseRepresentations(words, wordsSize);
    return 0;
}